linker/Mono.Linker.Steps/SweepStep.cs

```diff
@@ -175,6 +175,8 @@ bool ResolveAllTypeReferences (AssemblyDefinition assembly)
 			foreach (TypeReference tr in assembly.MainModule.GetTypeReferences ()) {
 				if (hash.ContainsKey (tr))
 					continue;
+				if (!CanChangeScope (tr))
```

Comment:
@marek-safar I looked into creating a "ChangeTypeScope" method and it gets kind of ugly.

First you have the `changes` variable to deal with.  Would probably need to become an out param.

Next there's the `hash` dictionary.  That would have to be passed in.

And lastly, there's the question of what to do with the second scope changing loop for modules with HasExportedTypes.  That loop has almost the same code, but it's not exactly the same.   Different enough that a `ChangeTypeScope` method might become ugly if we tried to reuse it in both places.

I'm not against creating a `ChangeTypeScope`, but I'm not going to go forward with it until you confirm that's the path you want to go down.  This `CanChangeType`  approach has it's advantages, low risk, we don't have to do any code duplication.